src/components/hmi_message_handler/test/CMakeLists.txt

```diff
@@ -38,10 +38,7 @@ include_directories (
 )
 
 if (HMIADAPTER STREQUAL "messagebroker")
-  set (BROKER_LIBRARIES
-    message_broker_client
-    message_broker_server
-  )
+  add_dependencies("hmi_message_handler_test" Boost)
```

Comment:
cmake fails on this line with HMIADAPTER set to `messagebroker`:
```
CMake Error at src/components/hmi_message_handler/test/CMakeLists.txt:41 (add_dependencies):
  Cannot add target-level dependencies to non-existent target
  "hmi_message_handler_test".

  The add_dependencies works for top-level logical targets created by the
  add_executable, add_library, or add_custom_target commands.  If you want to
  add file-level dependencies see the DEPENDS option of the add_custom_target
  and add_custom_command commands.

```
This line probably isn't even necessary.